### PR TITLE
Axios instance method supports generic

### PR DESCRIPTION
**Pull request by vietanhtran16**
Describe your pull request here.
- Add generic support for Axios instance method

### CODE REVIEW

File: test/typescript/axios.ts [hunk 1]
```diff
@@ -130,7 +138,7 @@ axios.get<User>('/user', { params: { id: 12345 } })
 axios.head<User>('/user')
 	.then(handleUserResponse)
     .catch(handleError);
-    
+
```
Comment (lucasrainett) [inline test/typescript/axios.ts:141]: If this is not related to your change, maybe don't include this, to keep the PR simpler.

**Review by lucasrainett — APPROVED** [review]

File: test/typescript/axios.ts (hunk 1 above)
Comment (vietanhtran16) [reply test/typescript/axios.ts:141]: thanks Lucas. I will probably move everything to a new branch so it would only have commit related to the type changes
